src/audio/codec_adapter/codec/generic.c

```diff
@@ -172,6 +172,47 @@ void *codec_allocate_memory(struct comp_dev *dev, uint32_t size,
 	return ptr;
 }
 
+int codec_free_memory(struct comp_dev *dev, void *ptr)
```

Comment:
So we're keeping a linked list of allocations so we can clean up just in case the codec requests a stupid deallocation or fails to clean up everything?

Reply:
Yes, codecs may not keep references of memory you gave them and/or don't provide API calls to deallocate that memory. Hence we let codec allocate memory using our interface so we keep eyes on it.